docs/website/docs/reference/frequently-asked-questions.md

```diff
@@ -0,0 +1,114 @@
+p = dlt.pipeline(
+    pipeline_name="my_pipeline",
+    destination="duckdb",
+    dataset_name="my_dataset"
+)
+
+# Extract data using the predefined source `my_source`
+p.extract(my_source().add_limit(10))
+
+# Normalize the data structure for consistency
+p.normalize()
+
+# Print the default schema of the pipeline in pretty YAML format for review
+print(p.default_schema.to_pretty_yaml())
+```
+
+This method ensures you obtain the full schema details, including all columns, as seen by `dlt`, without needing a predefined contract on these resources.
+
+## Can `dlt` periodically commit updates to a resource state to an external database during a pipeline run?
+
+Currently, `dlt` does not offer the functionality to checkpoint the extraction process during its execution. This limitation means that any updates to a resource state are not committed until the pipeline has fully completed its run. This issue has been recognized by the `dlt` development team and is currently being tracked under a specific GitHub issue ([#215](https://github.com/dlt-hub/dlt/issues/215)). Unfortunately, there is no specified timeline for when this feature will be available.
+
+In light of this limitation, `dlt` advises a workaround for incremental loading. This involves specifying both the start and end dates or page numbers for your data extraction process and conducting the backfill using smaller, manageable ranges, which are usually divided into around 100 loads. Implementing this strategy provides a higher degree of control and significantly mitigates the risk of losing any progress should the pipeline experience a failure or if it needs to be stopped midway through execution. For a more comprehensive understanding of how to implement this workaround, it is recommended to consult `dlt's` documentation on incremental loading with a cursor field, which can be found here: [Incremental Loading with a Cursor Field](../../docs/general-usage/incremental-loading#incremental-loading-with-a-cursor-field).
+
+## Is truncating or deleting a staging table safe?
+
+You can safely truncate those or even drop the whole staging dataset. However, it will have to be recreated on the next load and might incur extra loading time or cost.
+
+## How can I develop a "custom" pagination tracker?
+
+There are two ways. One, you can use `dlt.sources.incremental` to create a custom cursor for tracking pagination in data streams that lack an explicit cursor field.
+
+Second, while it's possible to utilize `dlt.sources.incremental` for tracking changes in data streams, you're not limited to this method for managing pagination or custom cursors. Instead, you can leverage the flexibility of managing state directly in Python. Access and modify the state like a standard Python dictionary: `state = dlt.current.resource_state(); state["your_custom_key"] = "your_value"`. This approach allows you to define your custom pagination logic based on your specific needs.
```

Comment:
add please the link to the example: https://github.com/dlt-hub/verified-sources/blob/master/sources/chess/__init__.py#L95

Reply:
Added